src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -37,8 +37,18 @@ public function getConfig()
         return $this->config;
     }
 
+    /**
+     * @deprecated since version 3.3, will be removed in 4.0. Use {@link getListeners()} instead.
+     */
     public function getContext()
     {
+        @trigger_error(sprintf('Method %s() is deprecated since version 3.3 and will be removed in 4.0. Use %s::getListeners() instead.', __METHOD__, __CLASS__), E_USER_DEPRECATED);
+
+        return $this->getListeners();
+    }
+
+    public function getListeners()
```

Comment:
this is a bit confusing considering that this returns both an array of listeners and an exception listener. ``getListeners`` may be expected to return just ``$this->listeners``